CMakeLists.txt

```diff
@@ -68,103 +51,222 @@ if(CMAKE_SYSTEM_PROCESSOR MATCHES "(x86)|(X86)|(amd64)|(AMD64)")
+    set_target_properties(D3D12Core PROPERTIES
+        IMPORTED_LOCATION "${GFX_AGILITY_PATH}/build/native/bin/x64/D3D12Core.dll"
+        IMPORTED_IMPLIB d3d12.lib
+        INTERFACE_INCLUDE_DIRECTORIES "${GFX_AGILITY_PATH}/build/native/include"
+    )
+    add_library(d3d12SDKLayers SHARED IMPORTED)
+    set_target_properties(d3d12SDKLayers PROPERTIES
+        IMPORTED_LOCATION "${GFX_AGILITY_PATH}/build/native/bin/x64/d3d12SDKLayers.dll"
+        IMPORTED_IMPLIB d3d12.lib
+    )
+    target_link_libraries(gfx PRIVATE D3D12Core d3d12SDKLayers)
+else()
+    find_package(directx-headers CONFIG REQUIRED)
+    target_link_libraries(gfx PRIVATE Microsoft::DirectX-Headers Microsoft::DirectX12-Agility)
+endif()
+
+set(GFX_DXC_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/directx-dxc")
+FetchContent_Declare(
+    directx-dxc
+    URL            "https://github.com/microsoft/DirectXShaderCompiler/releases/download/v1.7.2212.1/dxc_2023_03_01.zip"
+    SOURCE_DIR     "${GFX_DXC_PATH}"
+    FIND_PACKAGE_ARGS NAMES directx-dxc
 )
+FetchContent_MakeAvailable(directx-dxc)
+if(NOT directx-dxc_FOUND)
+    find_library(D3D12_LIB NAMES d3d12)
+    add_library(dxcompiler SHARED IMPORTED)
+    set_target_properties(dxcompiler PROPERTIES
+        IMPORTED_LOCATION "${GFX_DXC_PATH}/bin/x64/dxcompiler.dll"
+        IMPORTED_IMPLIB "${GFX_DXC_PATH}/lib/x64/dxcompiler.lib"
+        INTERFACE_INCLUDE_DIRECTORIES "${GFX_DXC_PATH}/inc"
+    )
+    add_library(dxil SHARED IMPORTED)
+    set_target_properties(dxil PROPERTIES
+        IMPORTED_LOCATION "${GFX_DXC_PATH}/bin/x64/dxil.dll"
+        IMPORTED_IMPLIB "${GFX_DXC_PATH}/lib/x64/dxcompiler.lib"
+    )
+    target_link_libraries(gfx PRIVATE dxcompiler dxil)
+else()
+    target_link_libraries(gfx PRIVATE Microsoft::DirectXShaderCompiler)
+endif()
 
+set(GFX_PIX_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/WinPixEventRuntime")
+set(GFX_PIX_AMD_PATH "${CMAKE_CURRENT_SOURCE_DIR}/third_party/AmdDxExt")
+FetchContent_Declare(
+    WinPixEventRuntime
+    URL            "https://www.nuget.org/api/v2/package/WinPixEventRuntime/1.0.240308001"
+    SOURCE_DIR     "${GFX_PIX_PATH}"
+)
+FetchContent_MakeAvailable(WinPixEventRuntime)
+file(READ "${GFX_PIX_PATH}/include/WinPixEventRuntime/PIXEvents.h" PIXE_FILE_CONTENTS)
+string(REPLACE "#define PIX_CONTEXT_EMIT_CPU_EVENTS" "# define PIX_CONTEXT_EMIT_CPU_EVENTS\n# include \"AmdPix3.h\"\n# define PIX_AMD_EXT\n" PIXE_FILE_CONTENTS "${PIXE_FILE_CONTENTS}")
+string(REPLACE " PIXBeginEventOnContextCpu(destination" " RgpPIXBeginEventOnContextCpu(destination" PIXE_FILE_CONTENTS "${PIXE_FILE_CONTENTS}")
+string(REPLACE " PIXSetMarkerOnContextCpu(destination" " RgpPIXSetMarkerOnContextCpu(destination" PIXE_FILE_CONTENTS "${PIXE_FILE_CONTENTS}")
+string(REPLACE "destination = PIXEndEventOnContextCpu(" "RgpPIXEndEventOnContextCpu(destination, " PIXE_FILE_CONTENTS "${PIXE_FILE_CONTENTS}")
+file(WRITE "${GFX_PIX_PATH}/include/WinPixEventRuntime/PIXEvents.h" "${PIXE_FILE_CONTENTS}")
```

Comment:
Auto patches WinPix with AMDPix3 on download, theres now no need to do it manually every time pix is updated

Reply:
That's neat.
Of course, this'd break if the patching process ever changes in a future version of Pix?
Not a problem tho, we can update that snippet then.

Reply:
Yep, would need to be changed if the function names changed. We have the winpix version in the cmake file so whenever bumping that we would have to change the patching process if needed. This should be pretty rare though